### PR TITLE
.gitignore: Readd `/tmp/`

### PR DESCRIPTION
**Pull request by Centril**
Specifically, `/tmp/partitioning-tests/` it is generated by the incremental tests, https://github.com/rust-lang/rust/search?p=2&q=partitioning-tests&unscoped_q=partitioning-tests. These are cleaned up by compiletest but not if you kill testing prematurely (which I just did to test out a rollup, and it is annoying to `rm -rf tmp/`).

r? @Mark-Simulacrum 
cc @RalfJung

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,6 +38,7 @@ __pycache__/
 /src/libcore/unicode/UnicodeData.txt
 /src/libcore/unicode/downloaded
 /target/
+/tmp/
```
Comment (RalfJung) [inline .gitignore]: Does gitignore supports comments?
```
/tmp/ # generated by ...
```

Comment (Centril) [reply .gitignore]: Seems so; https://stackoverflow.com/a/8865858/1063961

Comment (Centril) [reply .gitignore]: Left a comment :)

Comment (RalfJung) [reply .gitignore]: Seems like only comments on their own lines work though. :/

**Comment by RalfJung** [comment]
Would be great if you could also add a comment for https://github.com/rust-lang/rust/pull/63373#discussion_r312061198 :)

**Comment by Centril** [comment]
@RalfJung Done :)

**Review by RalfJung — APPROVED** [review]
LGTM. I'd say r=me but I feel this is outside my jurisdiction.^^

**Comment by Mark-Simulacrum** [comment]
@bors r+

I think .gitignore is both outside and inside everyone's jurisdiction.. tragedy of the commons and all.

**Comment by petrochenkov** [comment]
FYI, there's an issue about removing this `tmp` - https://github.com/rust-lang/rust/issues/34586.

**Comment by Centril** [comment]
@petrochenkov Thanks for the cc; Let's remove it again when it isn't produced anymore :)

**Pull request merged**